In [1]:
import pandas as pd
from sklearn.model_selection import GridSearchCV
from sklearn.ensemble import RandomForestClassifier

In [2]:
df = pd.read_csv('Encoded_Loan_Data.csv')

In [6]:
X = df[['Gender_Encode','Married_Encode','Education_Encode','Self_Employed_Encode', 'Dependents', 'ApplicantIncome',
       'CoapplicantIncome', 'LoanAmount', 'Loan_Amount_Term', 'Credit_History','Property_Area_Encode']]

y = df['Loan_Status_Encode']

In [28]:
from sklearn.model_selection import train_test_split

In [29]:
X_train, X_test, y_train, y_test= train_test_split(X,y,test_size=0.3,random_state=42)

In [12]:
params = {'max_depth':range(1,7),'min_samples_leaf':range(1,6),'n_estimators':range(25,201,25)}

In [30]:
grid_rfc = GridSearchCV(estimator=RandomForestClassifier(),param_grid=params,cv=5,verbose=2,n_jobs=3)
grid_rfc.fit(X_train,y_train)


Fitting 5 folds for each of 240 candidates, totalling 1200 fits


GridSearchCV(cv=5, estimator=RandomForestClassifier(), n_jobs=3,
             param_grid={'max_depth': range(1, 7),
                         'min_samples_leaf': range(1, 6),
                         'n_estimators': range(25, 201, 25)},
             verbose=2)

In [33]:
grid_rfc.best_params_

{'max_depth': 2, 'min_samples_leaf': 2, 'n_estimators': 125}

In [34]:
grid_rfc.best_estimator_

RandomForestClassifier(max_depth=2, min_samples_leaf=2, n_estimators=125)

In [35]:
print('Score of Random Forest Algorithm is',round((grid_rfc.best_score_) * 100,2))

Score of Random Forest Algorithm is 81.35


In [46]:
y_pred = grid_rfc.predict(X_test)

In [51]:
from sklearn.metrics import accuracy_score,precision_score,recall_score,f1_score

In [53]:
print('Accuracy score on test data',round(accuracy_score(y_test,y_pred)*100,2))
print('Precision score on test data',round(precision_score(y_test,y_pred)*100,2))
print('Recall score on test data',round(recall_score(y_test,y_pred)*100,2))
print('F1 score on test data',round(f1_score(y_test,y_pred)*100,2))

Accuracy score on test data 79.44
Precision score on test data 78.06
Recall score on test data 97.58
F1 score on test data 86.74


<h3>Training Data Score</h3>
<p>Training data Accuracy Score of Logistic Regression is 81.34</p>
<p>Training data Accuracy Score of KNN model is 79.43</p>
<p>Training data score of SVM model is 82.06</p>
<p>Training data score of Random Forest is 81.35</p>

<h3>Testing Data Score</h3>
<p>Testing data Accuracy Score of Logistic Regression is 79.44</p>
<p>Testing data Accuracy Score of KNN model is 75.56</p>
<p>Testing data score of SVM model is 80.0</p>
<p>Testing data Score of Random Forest 79.44</p>